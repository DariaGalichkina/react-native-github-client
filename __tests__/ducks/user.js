
import mockStore from 'redux-mock-store';

import reducer, { getUser, setUserData } from '../../app/ducks/user.js';

describe('User # reducer', () => {

  it('is expected to return appropriate state', () => {
    expect(reducer(null, setUserData({ login: 'login' }))).toMatchSnapshot();
  });

});

describe('User # async/getUser', () => {

  afterEach(() => nock.cleanAll())

  it('is expected to trigger set data action with received data', async () => {

    nock('https://api.github.com')
      .get('/user')
      .reply(200, { name: 'Superman' });

    const store = mockStore({ user: null });

    await store.dispatch(getUser());

    expect(store.getActions()).toMatchSnapshot();

  });

});

describe('User # setUserData', () => {

  it('is expected to create SET_DATA action', () => {
    expect(setUserData({ name: 'some' })).toMatchSnapshot();
  });

});
