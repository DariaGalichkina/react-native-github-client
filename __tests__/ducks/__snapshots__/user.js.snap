// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User # async/getUser is expected to trigger set data action with received data 1`] = `
Array [
  Object {
    "type": "user/SET_DATA",
    "userData": Object {
      "name": "Superman",
    },
  },
]
`;

exports[`User # reducer is expected to return appropriate state 1`] = `
Object {
  "login": "login",
}
`;

exports[`User # setUserData is expected to create SET_DATA action 1`] = `
Object {
  "type": "user/SET_DATA",
  "userData": Object {
    "name": "some",
  },
}
`;
